### ISSUE TITLE
Issue with updating position for roller blinds causing child bridge to crash.

### ISSUE THREAD

**Issue by tjax2017** [gormanb/homebridge-connector-hub#15]
**Describe The Bug:**
Getting a type error with the plugin which restarts the homebridge child bridge potentially due to reading the current position of Dooya roller blinds.
 
**To Reproduce:**

1. Add the driver
2. Enter the Connector+ key
3. Add the connector hub IPv4 address
4. Start the plugin with Homebridge
5. Plugin loads as a child bridge, connects, and discovers the 28 blinds but then fails to receive the blind positions and crashes the child bridge.

- I have tried two platforms - Docker and Synology - no change.
- I have tried 1.5 - 1.9 versions of the driver - no change
- I tried the Beta version - this time I get an authentication error.

**Expected behaviour:**
Blind positions are read and the Homebridge service displays the blinds and battery stats.

**Logs:**

```
[7/28/2023, 10:42:37 AM] [homebridge-connector-hub] Failed to get position: Connector Device 500291e4cf96001c
TypeError: Cannot read properties of undefined (reading 'currentPosition')
    at ConnectorAccessory.updateDeviceStatus (/volume1/homebridge/node_modules/homebridge-connector-hub/src/connectorAccessory.ts:154:43)
[7/28/2023, 10:42:40 AM] [homebridge-connector-hub] Child bridge process ended
[7/28/2023, 10:42:40 AM] [homebridge-connector-hub] Process Ended. Code: 1, Signal: null```
```

**Plugin Config:**
```
{
    "hubIps": [
        "192.xxx.86.158"
    ],
    "connectorKey": "eaxxxbe-xxxx-41",
    "enableDebugLog": true,
    "_bridge": {
        "username": "0E:xx:AA:CF:AB:xx",
        "port": 39163
    },
    "platform": "ConnectorHub"
}
```

**Screenshots:**
![image](https://github.com/gormanb/homebridge-connector-hub/assets/26449861/40576e0f-4cbd-485d-b2c6-8b3bb283ca8c)

**Environment:**

* **Plugin Version**: 1.0.9
* **Homebridge Version**: 1.6.1
* **Node.js Version**: 18.15.0
* **NPM Version**: 
* **Operating System**: Synology and tested with Docker
* **Connector Version**: 2.9.7
* **Hardware**: Dooya Roller Blinds x 28 with the Mini Hub 1554

**Comment by gormanb** [comment]
Hi @tjax2017 - from your message above, it seems that the plugin is receiving a response from the hub that does not contain the standard, expected message contents; specifically, [on this line](https://github.com/gormanb/homebridge-connector-hub/blob/115c339bb0e2566a2dbf135fee8b8114c69f1b45/src/connectorAccessory.ts#L154) it appears that the `data` field does not exist. Can you try installing the latest release `1.1.0` and turn on debug logging? I've included extra logging which should show the contents of this malformed message before the crash.

**Comment by gormanb** [comment]
Hi @tjax2017 - I've just published `1.1.1`, which should handle this exception without crashing the bridge. I was able to reproduce the problem locally; it appears to be due to an invalid Connector Key. When the plugin attempt to read the blinds' state on startup, they provide an Access Token as part of the request. This Access Token is computed from the Connector Key and the Token that was previously obtained from the hub. If the Connector Key is invalid, then the Access Token will also be invalid, and the request will be rejected by the hub. If you install `1.1.1` you should see an error message that looks something like this:
```
[8/7/2023, 10:18:28 AM] [ConnectorHub] Error received from hub. App key may be invalid: {
  msgType: 'WriteDeviceAck',
  mac: <mac>,
  deviceType: '10000000',
  msgID: '20230807101819579',
  token: <token>,
  actionResult: 'AccessToken error'
}
```
Please give `1.1.1` a try to confirm whether this is in fact the problem. If so, please make sure that you've entered the Connector Key in the plugin config exactly as it appears in the app.

**Comment by tjax2017** [comment]
Bernard,

100% working now.  I just updated the plugin, and it all worked perfectly.  I also updated the version of the firmware on the mini hub (once I found out how to do this).  I think the firmware needed to be updated as I think it was quite old > 12 months old.

Thank you so much for doing this, it is a great driver and adds so much to the homebridge ecosystem given these blinds are commonplace and relatively inexpensive (compared to Somfy etc) your work enables a great many of us to automate our homes (and annoy our wives with automation).

I think the only bit needed now would be the ability to read the battery % and trigger some automation to remind me to charge the blinds!

Regards

Tim

**Comment by gormanb** [comment]
Hi @tjax2017 - I'm delighted to hear that your blinds are working as expected now, and thanks for the kind words! Unfortunately I don't believe it's possible to create an automation based on the blinds' battery level, but a "low battery" indicator will show up on the Home app when they fall to below 20%.

Closing this issue as it is fixed in commit d355f5b.
